src/udev/udev-builtin-net_id.c

```diff
@@ -349,6 +352,50 @@ static int names_vio(struct udev_device *dev, struct netnames *names) {
         return 0;
 }
 
+#define _PLATFORM_TEST "/sys/devices/platform/vvvvPPPP"
+#define _PLATFORM_PATTERN4 "/sys/devices/platform/%4s%4x:%2x/net/eth%u"
+#define _PLATFORM_PATTERN3 "/sys/devices/platform/%3s%4x:%2x/net/eth%u"
+
+static int names_platform(struct udev_device *dev, struct netnames *names, bool test) {
+        struct udev_device *parent;
+        char vendor[5];
+        char shortid;
+        unsigned model, instance, ethid;
+        const char *syspath, *pattern;
+
+        /* check if our direct parent is a platform device with no other bus in-between */
+        parent = udev_device_get_parent(dev);
+        if (!parent)
+                return -ENOENT;
+
+        if (!streq_ptr("platform", udev_device_get_subsystem(parent)))
+                 return -ENOENT;
+
+        syspath = udev_device_get_syspath(dev);
+
+        /* Vendor ID inside ACPI id can be either 3 or 4 */
+        if (syspath[sizeof _PLATFORM_TEST - 1] == ':')
```

Comment:
uh, you never check whether syspath actually is long enough for checking this character...